### PR TITLE
feat: only enable rcmgr by default in full nodes

### PR DESCRIPTION
**Pull request by vyzo**
opt-in with envvar for other nodes, as there are still some issues with markets related protocols

### CODE REVIEW

**Comment by vyzo** [comment]
Needs backport to nv16 branch for v1.16.0 release.

File: node/modules/lp2p/rcmgr.go [hunk 1]
```diff
@@ -27,9 +27,10 @@ import (
 
 func ResourceManager(connMgrHi uint) func(lc fx.Lifecycle, repo repo.LockedRepo) (network.ResourceManager, error) {
 	return func(lc fx.Lifecycle, repo repo.LockedRepo) (network.ResourceManager, error) {
+		isFullNode := repo.RepoType().Type() == "FullNode"
 		envvar := os.Getenv("LOTUS_RCMGR")
-		if envvar == "0" {
-			// this is enabled by default; specify LOTUS_RCMGR=0 to disable
+		if (isFullNode && envvar == "0") || // enable by default for full nodes
```
Comment (ribasushi) [inline node/modules/lp2p/rcmgr.go:32]: Logic seems wrong, I think it should be:
```suggestion
		if (isFullNode && envvar != "0") || // enable by default for full nodes
```

Comment (vyzo) [reply node/modules/lp2p/rcmgr.go:32]: we might want to fix the comment to avoid tripping casual reading.
Any suggestions?

Comment (ribasushi) [inline node/modules/lp2p/rcmgr.go:32]:
```suggestion
		if (isFullNode && envvar == "0") || // only set NullResourceManager if envvar is explicitly "0"
```

File: node/modules/lp2p/rcmgr.go [hunk 2]
```diff
@@ -27,9 +27,10 @@ import (
...
+			(!isFullNode && envvar != "1") { // otherwise, opt-in with envvar
```
Comment (ribasushi) [inline node/modules/lp2p/rcmgr.go:33]:
```suggestion
			(!isFullNode && envvar != "1") { // set NullResourceManager *unless* envvar is explicitly "1"
```

Comment (ribasushi) [reply node/modules/lp2p/rcmgr.go:33]: @vyzo ☝️ maybe

Comment (vyzo) [reply node/modules/lp2p/rcmgr.go]: thanks, applied.

**Review by magik6k — APPROVED** [review]

**Pull request merged**